src/displacement.m

```diff
@@ -24,39 +24,40 @@
        return
     end
     
-    N_cell = count_nearest_neighbours(k,l,size(system.age,1),size(system.age,2));
-    
     state_ = system.state(k,l);
     
-    id_lin = [k-1,k,k+1];
-    id_col = [l-1,l,l+1];
+    [id_lin, id_col] = nearest_neighbours(k,l,N_lin,N_col);
+    
+    N_cell = count_nearest_neighbours(k,l,N_lin,N_col);
+    
+    N_lin_nn = size(id_lin,2);
+    N_col_nn = size(id_col,2);
     
     moved = false;
     
     if(state_ == 'I' || state_ == 'R')
         
         p = rand;
+        
+        %probability to change cell
         q = 0.5/N_cell;

```

Comment:
Maybe you want to change that number in order to be able to vary the number 0.5 ?
What about:
`
q = system.cfg.proba_displacement / N_cell;
`
If you say it is not relevant it’s okay